lib/contracts/formatters.rb

```diff
@@ -85,8 +85,28 @@ def plain?
 
       def useful_to_s?
         # Useless to_s value or no custom to_s behavious defined
-        # Ruby < 2.0 makes inspect call to_s so this won't work
-        @value.to_s != "" && @value.to_s != @value.inspect
+        !empty_to_s? && custom_to_s?
+      end
+
+      def empty_to_s?
+        @value.to_s.empty?
+      end
+
+      def custom_to_s?
+        if @value.class == Class # It's a class contract
```

Comment:
```ruby
if @value.class == Class
  # ...
else
  # ...
end
```

This construction duplicated. Not that it is so bad, but could be better:

```ruby
class InspectWrapper
  def lifted_class
    @value.class == Class ? ClassContractInspectWrapper : InstanceContractInspectWrapper
  end

  def lift
    lifted_class.new(@value)
  end

  def custom_to_s?
    lift.custom_to_s?
  end

  def raw_inspect
    lift.raw_inspect.gsub(/^Contracts::/, "")
  end
end

class InstanceContractInspectWrapper
  def custom_to_s?
    !@value.to_s.match(/#\<\w+:.+\>/)
  end

  def raw_inspect
    empty_to_s? ? @value.class.name : @value.inspect
  end
end

class ClassContractInspectWrapper
  def custom_to_s?
    @value.to_s != @value.name
  end

  def raw_inspect
    empty_to_s? ? @value.name : @value.inspect
  end
end
```

More complicated, but more DRY and robust.